internal/operands/common-templates/reconcile.go

```diff
@@ -0,0 +1,151 @@
+		&core.Namespace{},
+		&templatev1.Template{},
+	}
+}
+
+func (c *commonTemplates) WatchTypes() []runtime.Object {
+	return nil
+}
+
+func (c *commonTemplates) Reconcile(request *common.Request) error {
+	for _, f := range []func(*common.Request) error{
+		reconcileGoldenImagesNS,
+		reconcileViewRole,
+		reconcileViewRoleBinding,
+		reconcileEditRole,
+		reconcileTemplates,
+	} {
+		if err := f(request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *commonTemplates) Cleanup(request *common.Request) error {
+	objects := []controllerutil.Object{
+		newGoldenImagesNS(GoldenImagesNSname),
+		newViewRole(GoldenImagesNSname),
+		newViewRoleBinding(GoldenImagesNSname),
+		newEditRole(),
+	}
+	namespace := request.Instance.Spec.CommonTemplates.Namespace
+	for _, template := range templatesBundle {
+		template.ObjectMeta.Namespace = namespace
+		objects = append(objects, &template)
+	}
+	for _, obj := range objects {
+		err := request.Client.Delete(request.Context, obj)
+		if err != nil && !errors.IsNotFound(err) {
+			request.Logger.Error(err, fmt.Sprintf("Error deleting \"%s\": %s", obj.GetName(), err))
+			return err
+		}
+	}
+	return nil
+}
+
+func reconcileGoldenImagesNS(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newGoldenImagesNS(GoldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			newNS := newRes.(*core.Namespace)
+			foundNS := foundRes.(*core.Namespace)
+			foundNS.Spec = newNS.Spec
+		})
+}
+
+func reconcileViewRole(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newViewRole(GoldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			foundRole := foundRes.(*rbac.Role)
+			newRole := newRes.(*rbac.Role)
+			foundRole.Rules = newRole.Rules
+		})
+}
+
+func reconcileViewRoleBinding(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newViewRoleBinding(GoldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			newBinding := newRes.(*rbac.RoleBinding)
+			foundBinding := foundRes.(*rbac.RoleBinding)
+			foundBinding.Subjects = newBinding.Subjects
+			foundBinding.RoleRef = newBinding.RoleRef
+		})
+}
+
+func reconcileEditRole(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newEditRole(),
+		func(newRes, foundRes controllerutil.Object) {
+			newRole := newRes.(*rbac.ClusterRole)
+			foundRole := foundRes.(*rbac.ClusterRole)
+			foundRole.Rules = newRole.Rules
+		})
+}
+
+func reconcileTemplates(request *common.Request) error {
+	loadTemplates := func() {
+		var err error
+		filename := filepath.Join(bundleDir, "common-templates-"+Version+".yaml")
+		templatesBundle, err = readTemplates(filename)
+		if err != nil {
+			request.Logger.Error(err, fmt.Sprintf("Error reading from template file; Templates found \n%v", templatesBundle))
+			panic(err)
+		}
+		if len(templatesBundle) == 0 {
+			panic(templatesBundle)
```

Comment:
Please provide a clear error here that states that no templates were found

Reply:
Added